packages/containers/src/SelectObject/SelectObject.container.js

```diff
@@ -88,6 +92,38 @@ export function filter(
 	return items;
 }
 
+/**
+ * apply query on every elements, return them on a single list,
+ * @return item in items found with the id
+ * @param {Object} options {query, items, idAttr }
+ */
+export function filterAll(
+	items = new List(),
+	query = '',
+	{ nameAttr = 'name', onMatch = noop } = {},
+	currentPosition = 'root',
+) {
+	let results = new List();
+
+	if (query) {
+		items.forEach(item => {
+			const name = item.get(nameAttr, '');
+			const children = item.get('children', null);
+			if (name.toLowerCase().includes(query.toLowerCase())) {
+				onMatch(item);
+				results = results.push(item.set('currentPosition', currentPosition));
+			}
+			if (children) {
+				results = results.concat(
+					filterAll(children, query, { nameAttr }, `${currentPosition} > ${name}`),
+				);
+			}
```

Comment:
Can you cut this block  to be more readable ? More functionnal oriented ? What do u think about it ?

Reply:
Something like that ?

```javascript
const filtered = filterAll(children, query, { nameAttr }, `${currentPosition} > ${name}`);
results = results.concat(filtered);
```